### ISSUE TITLE
[Feature] Automatically create jobs if Spring Bean is annotated with @Job annotation

### ISSUE THREAD

**Issue by rdehuyss** [jobrunr/jobrunr#977]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I need to write boilerplate code. I'm not a fan of always repeating:

```java
jobScheduler.enqueue(() -> emailClient.send(to));
```

**Describe the solution you'd like**
The idea would be that you could annotate an interface and spring could create a proxy so that when you invoke a method on an implementation of that interface it schedules a job for you (to be run).

```java
@JobGateway // or @Async
interface EmailClient {
 @Job(name =“sendEmail”) 
 void send(String to); 
}
```

**Describe alternatives you've considered**
🤔 can't think of anything else...

**Additional context**
Reported by @joshlong

**Comment by devashishTaneja** [comment]
Hi @rdehuyss, is this feature open for contribution?

**Comment by rdehuyss** [comment]
Yes, of course. Feel free to ping me if you want to discuss ideas on how to implement this.

**Comment by devashishTaneja** [comment]
Hi @rdehuyss,

I have raised an initial PR for this. Please let me know if it looks good.

https://github.com/jobrunr/jobrunr/pull/1181

**Comment by rdehuyss** [comment]
Resolved in V8 branch. Thanks, @devashishTaneja!
